### PR TITLE
locker: use non-overflowing keys and tx scoping

### PR DESCRIPTION
**Pull request by tsenart**
This commit introduces two changes:

1. Non overflowing keys from int(64) to int32 which is what an int in
   Postgres maps to (bigint maps to int64).
2. Use transaction scoped advisory locks (i.e. `pg_advisory_xact_lock`).
   This results in `LockInTransaction` no longer returning an
   UnlockFunc, since the lock is released when the tx is rolled back or
   committed.

### CODE REVIEW

File: internal/database/locker/locker.go [hunk 1]
```diff
@@ -76,23 +75,27 @@ func (l *Locker) Lock(ctx context.Context, key int, blocking bool) (locked bool,
...
 	}()
 
-	locked, unlock, err := tx.LockInTransaction(ctx, key, blocking)
+	locked, err = tx.LockInTransaction(ctx, key, blocking)
 	if err != nil || !locked {
 		return false, nil, err
 	}
 
-	return true, func(err error) error { return tx.Done(unlock(err)) }, nil
+	return true, func(err error) error { return tx.Done(err) }, nil
```
Comment (efritz) [inline internal/database/locker/locker.go:83]: You can just turn tx.Done here

**Review by efritz — APPROVED** [review]
Great improvement!

File: internal/database/locker/locker.go (hunk 1 above)
Comment (tsenart) [reply internal/database/locker/locker.go:83]: Nice!

Comment (tsenart) [inline internal/database/locker/locker.go:83]:
```suggestion
	return true, tx.Done, nil
```

**Review by mrnugget — APPROVED** [review]
Good stuff!

Side-note to @eseliger: looks like we have our answer now regarding locks and transactions :smile:

**Pull request merged**